IaC/PnP/monitoring/aks/online/aks_secure_baseline/02-aks.md

```diff
@@ -0,0 +1,158 @@
+  metadata:
+      name: podmi-ingress-controller-identity
+      namespace: a0008
+  spec:
+      type: 0
+      resourceID: $TRAEFIK_USER_ASSIGNED_IDENTITY_RESOURCE_ID
+      clientID: $TRAEFIK_USER_ASSIGNED_IDENTITY_CLIENT_ID
+  ---
+  apiVersion: aadpodidentity.k8s.io/v1
+  kind: AzureIdentityBinding
+  metadata:
+      name: podmi-ingress-controller-binding
+      namespace: a0008
+  spec:
+      azureIdentity: podmi-ingress-controller-identity
+      selector: $INGRESS_IDENTITY
+  EOF
+  ```
+
+1. Create the Traefik's Secret Provider Class resource.
+
+   > The Ingress Controller will be exposing the wildcard TLS certificate you created in a prior step. It uses the Azure Key Vault CSI Provider to mount the certificate which is managed and stored in Azure Key Vault. Once mounted, Traefik can use it.
+   >
+   > Create a `SecretProviderClass` resource with with your Azure Key Vault parameters for the [Azure Key Vault Provider for Secrets Store CSI driver](https://github.com/Azure/secrets-store-csi-driver-provider-azure).
+
+    ```bash
+    KEYVAULT_NAME=$(terraform output -json | jq -r .keyvaults.value.secrets.name)
+    TENANTID_AZURERBAC=$(az account show --query tenantId -o tsv)
+    ```
+    ```yaml
+    cat <<EOF | kubectl apply -f -
+    apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+    kind: SecretProviderClass
+    metadata:
+      name: aks-ingress-ngsa-monitoring-dev-com-tls-secret-csi-akv
+      namespace: a0008
+    spec:
+      provider: azure
+      parameters:
+        usePodIdentity: "true"
+        keyvaultName: $KEYVAULT_NAME
+        objects:  |
+          array:
+            - |
+              objectName: wildcard-ingress
+              objectAlias: tls.crt
+              objectType: cert
+            - |
+              objectName: wildcard-ingress
+              objectAlias: tls.key
+              objectType: secret
+        tenantId: $TENANTID_AZURERBAC
+    EOF
+
+2. Update Traefik config to pin IP in Aks-ingress Subnet:
+    ```bash
+    # Get the ingress controller subnet name
+    ingress_subnet_name=$(terraform output -json | jq -r .vnets.value.vnet_aks_re1.subnets.aks_ingress.name)
+    # Update the traefik yaml
+    # Mac UNIX: 
+    sed -i "" "s/azure-load-balancer-internal-subnet:.*/azure-load-balancer-internal-subnet:\ ${ingress_subnet_name}/g" online/aks_secure_baseline/workloads/baseline/traefik.yaml
+
+    # Linux:
+    sed -i "s/azure-load-balancer-internal-subnet:.*/azure-load-balancer-internal-subnet:\ ${ingress_subnet_name}/g" online/aks_secure_baseline/workloads/baseline/traefik.yaml
+    ```
+
+3. Deploy Traefik & ASP.net sample appplication
+    ```bash
+    kubectl apply -f online/aks_secure_baseline/workloads/baseline
+    # It takes 2-3 mins to deploy Traefik & the sample app. Watch all pods to be provision with:
+    kubectl get pods -n a0008 -w
+    # Ensure sample app ingress has IP assigned
+    kubectl get ingress -n a0008
+    # This website will be available at the public domain below
```

Comment:
We need to remind the user to update the etc/hosts file, as the website is not yet in the public domain

Reply:
@PurpleBriar done

Reply:
Also, once we lock on external domain strategy, this should be updated again so the hosts file change is not necessary.

Reply:
comment added to markdown